doc/namespace-store-crd.md

```diff
@@ -0,0 +1,218 @@
+spec:
+  azureBlob:
+    secret:
+      name: namespace-store-azure-blob-bs
+      namespace: noobaa
+    targetBlobContainer: CONTAINER
+  type: azure-blob
+```
+
+#### Credentials change
+
+In case the credentials of a Namespace-store need to be updated due to a periodic security policy or concern, the appropriate secret should be updated by the user, and the operator will be responsible for watching changes in those secrets and propagating the new credential update to the NooBaa system server.
+
+
+# Read Status
+
+Here is an example healthy status (see below example of non-healthy status):
+
+```yaml
+apiVersion: noobaa.io/v1alpha1
+kind: NamespaceStore
+metadata:
+  name: aws-s3
+  namespace: noobaa
+spec:
+  ...
+status:
+  conditions:
+  - lastHeartbeatTime: "2019-11-05T13:50:50Z"
+    lastTransitionTime: "2019-11-06T07:03:46Z"
+    message: noobaa operator completed reconcile - namespace store is ready
+    reason: NamespaceStorePhaseReady
+    status: "True"
+    type: Available
+  - lastHeartbeatTime: "2019-11-05T13:50:50Z"
+    lastTransitionTime: "2019-11-06T07:03:46Z"
+    message: noobaa operator completed reconcile - namespace store is ready
+    reason: NamespaceStorePhaseReady
+    status: "False"
+    type: Progressing
+  - lastHeartbeatTime: "2019-11-05T13:50:50Z"
+    lastTransitionTime: "2019-11-05T13:50:50Z"
+    message: noobaa operator completed reconcile - namespace store is ready
+    reason: NamespaceStorePhaseReady
+    status: "False"
+    type: Degraded
+  - lastHeartbeatTime: "2019-11-05T13:50:50Z"
+    lastTransitionTime: "2019-11-06T07:03:46Z"
+    message: noobaa operator completed reconcile - namespace store is ready
+    reason: NamespaceStorePhaseReady
+    status: "True"
+    type: Upgradeable
+  phase: Ready
+```
+
+
+# Delete
+
+Namespace-Store are used for data persistency, therefore there is a cleanup process before they can be deleted.
+The operator will use the `finalizer` pattern as explained in the link below, and set a finalizer on every Namespace-store to mark that external cleanup is needed before it can be delete:
+
+https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers
+
+After marking a Namespace-store for deletion, the operator will notify the NooBaa server on the deletion which will enter a *decommissioning* state, in which NooBaa will attempt to rebuild the data to a new Namespace-store location. Once the decomissioning process completes the operator will remove the finalizer and allow the CR to be deleted.
+
+There are cases where the decommissioning cannot complete due to inability to read the data from the Namespace-store that is already not serving - for example if the target bucket was already deleted or the credentials were invalidated or there is no network from the system to the Namespace-store service. In such cases the system status will be used to report these issues and suggest manual resolution for example:
```

Comment:
```suggestion
Namespace-Store is used for data persistency, therefore there is a cleanup process before they can be deleted.
The operator will use the `finalizer` pattern as explained in the link below, and set a finalizer on every Namespace-store to mark that external cleanup is needed before it can be deleted:

https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers

After marking a Namespace-store for deletion, the operator will notify the NooBaa server on the deletion which will enter a *decommissioning* state, in which NooBaa will attempt to rebuild the data to a new Namespace-store location. Once the decommissioning process completes the operator will remove the finalizer and allow the CR to be deleted.

There are cases where the decommissioning cannot complete due to the inability to read the data from the Namespace-store that is already not serving - for example, if the target bucket was already deleted or the credentials were invalidated or there is no network from the system to the Namespace-store service. In such cases the system status will be used to report these issues and suggest a manual resolution for example:
```